### DISCUSSION TITLE
Running `npx nx serve` leads to using root package.json, not defaultProject

### DISCUSSION THREAD

**Discussion by Lonli-Lokli** [nrwl/nx#21022] [General]
When I am running npx nx serve it is trying to run serve with the name of root package.json, not the default app. What is the proper way to make default app?

**Comment by yzhe554** [comment]
I am wondering if you have `"nx": {},` in package.json, which will always try to run `root:serve`
Ref: https://nx.dev/recipes/running-tasks/root-level-scripts

I think it would be a good idea to introduce a flag to exclude scripts from root package.json in some cases.

The workaround I have is to create a script(e.g `serve:default`) in package.json and the script will run the default app. so `nx serve:default` is used.
